app/frontend/src/components/MarkdownInput/UploadImage.tsx

```diff
@@ -0,0 +1,65 @@
+import { Emitter } from "@toast-ui/editor";
+import Button from "components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+} from "components/Dialog";
+import ImageInput from "components/ImageInput";
+import TextField from "components/TextField";
+import {
+  IMAGE_UPLOAD_INPUT_ALT,
+  UPLOAD_IMAGE,
+} from "features/communities/constants";
+import React, { useRef } from "react";
+import { useForm } from "react-hook-form";
+import { ImageInputValues } from "service/api";
+
+interface UploadImageProps {
+  emitter?: Emitter;
+  open: boolean;
+  onClose: () => void;
+}
+
+export default function UploadImage({
+  emitter,
+  open,
+  onClose,
+}: UploadImageProps) {
+  const alt = useRef("");
+  const { control } = useForm();
+  const handleSuccess = async (data: ImageInputValues) => {
+    emitter?.emit("command", "addImage", {
+      imageUrl: data.full_url,
+      altText: alt.current,
```

Comment:
How are these being used by toast UI (or our code?) to display the image after?

Reply:
Magic, that's how

Reply:
(It just inserts the correct markdown `![alt text](url)`)

Reply:
Is said black magic even documented anywhere? Or is it a bit of digging into their source code?

Reply:
Indeed it is source code digging, the docs are pretty poor and worse since they are mostly not updated since v3 was released